autokeras/tasks/time_series_forecaster.py

```diff
@@ -1,7 +1,118 @@
+import pandas as pd
+
 from autokeras import auto_model
+from autokeras import hypermodels
+from autokeras import nodes as input_module
+from autokeras.tuners import greedy
+
+
+class SupervisedTimeseriesDataPipeline(auto_model.AutoModel):
+
+    def __init__(self,
+                 outputs,
+                 column_names=None,
+                 column_types=None,
+                 lookback=None,
+                 predict_from=1,
+                 predict_until=10,
+                 **kwargs):
+        inputs = input_module.TimeseriesInput(lookback=lookback,
+                                              column_names=column_names,
+                                              column_types=column_types)
+        if column_types:
+            for column_type in column_types.values():
+                if column_type not in ['categorical', 'numerical']:
+                    raise ValueError(
+                        'Column_types should be either "categorical" '
+                        'or "numerical", but got {name}'.format(name=column_type))
+        if column_names and column_types:
+            for column_name in column_types:
+                if column_name not in column_names:
+                    raise ValueError('Column_names and column_types are '
+                                     'mismatched. Cannot find column name '
+                                     '{name} in the data.'.format(name=column_name))
+        super().__init__(inputs=inputs,
+                         outputs=outputs,
+                         **kwargs)
+        self.predict_from = predict_from
+        self.predict_until = predict_until
+        self._target_col_name = None
+        self.train_len = 0
+
+    @staticmethod
+    def _read_from_csv(x, y):
+        df = pd.read_csv(x)
+        target = df.pop(y).dropna().to_numpy()
+        return df, target
+
+    def fit(self,
+            x=None,
+            y=None,
+            epochs=None,
+            callbacks=None,
+            validation_split=0.2,
+            validation_data=None,
+            **kwargs):
+        # x is file path of training data
+        if isinstance(x, str):
+            self._target_col_name = y
+            x, y = self._read_from_csv(x, y)
+        if validation_data:
+            x_val, y_val = validation_data
+            if isinstance(x_val, str):
+                validation_data = self._read_from_csv(x_val, y_val)
```

Comment:
This part can also be extracted into a function to share with StructuredData.

Reply:
self._read_from_csv(x, y) is different for StructuredData and Timeseries so I don't think we can put this piece into a mixin.